fuel-tx/src/transaction.rs

```diff
@@ -832,6 +832,27 @@ pub mod field {
         }
     }
 
+    pub trait Expiration {
+        fn expiration(&self) -> BlockHeight;
+        fn set_expiration(&mut self, value: BlockHeight);
+    }
+
+    impl<T: Policies + ?Sized> Expiration for T {
+        #[inline(always)]
+        fn expiration(&self) -> BlockHeight {
+            self.policies()
+                .get(PolicyType::Expiration)
+                .map(|value| u32::try_from(value).unwrap_or(u32::MAX).into())
+                .unwrap_or_default()
+        }
```

Comment:
I noticed that there are a few places you added:
```rs
.with_expiration(u32::MAX.into());
```

I feel like that shouldn't be necessary. Perhaps we should something like:
```suggestion
        fn expiration(&self) -> BlockHeight {
            self.policies()
                .get(PolicyType::Expiration)
                .and_then(|value| u32::try_from(value).ok()))
                .unwrap_or(u32::MAX)
                .into()
        }
```
Or something like that? i.e. move the `unwrap_or` up one level?

Reply:
Yeah I agree the default behavior is better if it doesn't constraint at all. Updated and updated all tests where it was not necessary.

Reply:
It doesn't specify in the specs, BTW. So we should probably clarify there.

Reply:
Will ask to our sync